integration_test/ops_agent_test.go

```diff
@@ -1440,19 +1442,54 @@ func TestTCPLog(t *testing.T) {
 		if err := agents.SetupOpsAgent(ctx, logger.ToMainLog(), vm, config); err != nil {
 			t.Fatal(err)
 		}
+		command := `echo '{"msg":"test tcp log 1"}{"msg":"test tcp log 2"}' | /opt/google-cloud-ops-agent/subagents/fluent-bit/bin/fluent-bit -i stdin -o tcp://127.0.0.1:5170 -p format=json_lines && echo '{"msg":"test tcp log 3"}{"msg":"test tcp log 4"}' | /opt/google-cloud-ops-agent/subagents/fluent-bit/bin/fluent-bit -i stdin -o tcp://127.0.0.1:5170 -p format=json_lines`
+		// Installing Netcat or equivalent tools on Windows is tricky so we can instead
+		// use this Powershell Script that can send TCP messages.
+		if gce.IsWindows(platform) {
+			tcp_script := `param([string]$TargetIP = "", [int]$TargetPort = 0, [string]$Message = "")
+
+			try {
+				if ($TargetIP -eq "" ) { $TargetIP = read-host "Enter target IP address" }
+				if ($TargetPort -eq 0 ) { $TargetPort = read-host "Enter target port" }
+				if ($Message -eq "" ) { $Message = read-host "Enter message to send" }
+
+					$IP = [System.Net.Dns]::GetHostAddresses($TargetIP) 
+					$Address = [System.Net.IPAddress]::Parse($IP) 
+					$Socket = New-Object System.Net.Sockets.TCPClient($Address,$TargetPort) 
+					$Stream = $Socket.GetStream() 
+					$Writer = New-Object System.IO.StreamWriter($Stream)
+					$Message | % {
+						$Writer.WriteLine($_)
+						$Writer.Flush()
+					}
+					$Stream.Close()
+					$Socket.Close()
+
+				"??  Done."
+				exit 0 # success
+			} catch {
+				"?? Error in line $($_.InvocationInfo.ScriptLineNumber): $($Error[0])"
+				exit 1
+			}`
+
+			if err := gce.UploadContent(ctx, logger.ToMainLog(), vm, strings.NewReader(tcp_script), `C:\Program Files\Google\Cloud Operations\Ops Agent\send_tcp.ps1`); err != nil {
+				t.Fatalf("Error uploading TCP script: %v", err)
+			}
+
+			command = `(& "C:\Program Files\Google\Cloud Operations\Ops Agent\send_tcp.ps1" 127.0.0.1 5170 '{"msg":"test tcp log 1"}{"msg":"test tcp log 2"}') -and (& "C:\Program Files\Google\Cloud Operations\Ops Agent\send_tcp.ps1" 127.0.0.1 5170 '{"msg":"test tcp log 3"}{"msg":"test tcp log 4"}')`
```

Comment:
WDYT about this version:

```suggestion
			tcpScript := `$ErrorActionPreference = 'Stop'
			function Send-TcpString {
			  param(
			    [string]$TargetIP,
			    [int]$TargetPort,
			    [string]$Message
			  )
			  $Socket = New-Object System.Net.Sockets.TCPClient($TargetIP, $TargetPort) 
			  $Stream = $Socket.GetStream() 
			  $Writer = New-Object System.IO.StreamWriter($Stream)
			  $Message | % {
			    $Writer.WriteLine($_)
			    $Writer.Flush()
			  }
			  $Stream.Close()
			  $Socket.Close()
			}
			Send-TcpString 127.0.0.1 5170 '{"msg":"test tcp log 1"}{"msg":"test tcp log 2"}'
			Send-TcpString 127.0.0.1 5170 '{"msg":"test tcp log 3"}{"msg":"test tcp log 4"}'`

			if _, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", tcpScript); err != nil {
				t.Fatalf("Error writing dummy TCP log line: %v", err)
			}
```

Rationale, mostly simplifications:
* Params should either have valid defaults or be required; may as well make them required here
* No need for intermediate `$IP` and `$Address` variables; the `TCPClient` constructor has an overload that takes a host and port directly
* You can define and invoke functions inline so we can skip the intermediate file upload
* The `Read-Host` stuff isn't needed since we aren't running an interactive prompt
* We've recently been trying to push for `$ErrorActionPreference = 'Stop'` -- the default error format is already sufficiently descriptive so we don't need to reformat it in a catch
* (nit) Use camelCase for local Go vars

Reply:
I fully agree that this is a much neater and better script in general! thank you for helping me with it!